### DISCUSSION TITLE
How is starship determining the python version?

### DISCUSSION THREAD

**Discussion by brandonleon** [starship/starship#6353] [Q&A] [answered]
I'm pretty sure this is an issue with my environment, but I just need a push in the right direction to figure out how Starship is getting the version.

![Screenshot 2024-10-24 at 3 34 54 PM](https://github.com/user-attachments/assets/686361ff-3fa6-419a-b588-4be111108842)

**Comment by davidkna** [comment]
With the default starship config, it will also execute `python --version`:

https://github.com/starship/starship/blob/master/src/modules/python.rs#L95-L104

**Reply by brandonleon** [reply]
Can I specify the python binary? How does it know which to run?

**Reply by davidkna** [reply, marked as answer]
As usual, executables are sourced from the executable `$PATH`. There is a `python_binary` option, but relative paths to python binaries are discouraged as they could lead to inadvertently executing untrusted files.

**Reply by brandonleon** [reply]
That answers my question. Here’s a bit more detail: the Python command is aliased to `uv run python`. This setup allows UV to detect the `.python-version` file and route commands to the correct version of Python. The downside is that `python` is not in my system $PATH, while `python3` is. However, `python3` is the Homebrew-installed version and is always 3.13. I created a [shell script](https://gist.github.com/brandonleon/f51d53efa04067fa9799130a13986170) that is not in my system $PATH and used python-path (absolute path) to ensure Starship uses it. This script passes the command to `uv run python`, and it now works as expected.

**Reply by fepegar** [reply]
In case it helps, this works for me:

```toml
[python]
python_binary = ".venv/bin/python"
```

Not sure what the problem is with relative paths.
